File: 0016-3sum-closest/three-pointer.cs
public class Solution
{
    public int ThreeSumClosest(int[] nums, int target)
    {
        int n = nums.Length;

        if (n < 3)
        {
            return 0;
        }
        else
        {
            // Sort array
            Array.Sort(nums);

            int result = nums[0] + nums[1] + nums[2];

            for (int i = 0; i < n; i++)
            {
                int left  = i + 1;
                int right = n - 1;

                while (left < right)
                {
                    int sum = nums[i] + nums[left] + nums[right];

                    int currentDifference = Math.Abs(sum - target);
                    int closestDifference = Math.Abs(result - target);

                    if (currentDifference < closestDifference) // Current sum is closer, thus update result
                    {
                        result = sum;
                    }

                    // Update pointers
                    if (sum < target)
                    {
                        left++;
                    }
                    else if (sum > target)
                    {
                        right--;
                    }
                    else
                    {
                        return target;
                    }
                }
            }

            // Default return
            return result;
        }
    }
}

File: 0016-3sum-closest/three-pointer.py
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        n = len(nums)

        if (n < 3):
            return 0
        else:
            # Sort array
            nums.sort()

            result = nums[0] + nums[1] + nums[2]

            for i in range(n):
                left  = i + 1
                right = n - 1

                while (left < right):
                    sum = nums[i] + nums[left] + nums[right]

                    currentDifference = abs(sum - target)
                    closestDifference = abs(result - target)

                    if (currentDifference < closestDifference):
                        result = sum
                    
                    # Update poitners
                    if (sum < target):
                        left += 1
                    elif (sum > target):
                        right -= 1
                    else:
                        return target
            
            # Default return
            return result
        